ocaml/testsuite/tests/typing-layouts-float64/parsing.ml

```diff
@@ -45,9 +45,9 @@ Line 1, characters 9-15:
              ^^^^^^
 Error: This type float# should be an instance of type ('a : value)
        The layout of float# is float64, because
-         it equals the primitive value type float#.
+         it equals the primitive float64 type float#.
```

Comment:
Upon reading this again, this message is a little silly. Of course `float# = float#`. Maybe

```
  it is a primitive float64 type.
```

Would be better. We'd have to make sure that

```
type myfloat = float#
```

would get the old message, though (with "equals").

I think this might be possible by comparing the path against the path for `float#`, as exported from predef.

Reply:
This seems a little difficult to implement. The path stored in the `Primitive` creation_reason is going to be `ident_unboxed_float` even in the case of `type myfloat = float#`. To enable this path check, `myfloat` will have to be threaded through all the layout history format functions to get to `format_float64_creation_reason`. There will also be a circular dependence between Predef and Layouts if we want the path of `float#` from Predef.

Not sure if all that will be worth it?

Reply:
Fine. Maybe just change `equals` to `is` and then we'll call it good enough. For `float#` the error message will b slightly pedantic. For synonyms, it will be useful.